daemon/algod/api/server/v2/utils.go

```diff
@@ -372,84 +366,79 @@ func ConvertInnerTxn(txn *transactions.SignedTxnWithAD) PreEncodedTxInfo {
 	return response
 }
 
-func convertScratchChanges(scratchChanges []simulation.ScratchChange) *[]model.ScratchChange {
-	if len(scratchChanges) == 0 {
-		return nil
+func convertToAVMValue(tv basics.TealValue) model.AvmValue {
+	return model.AvmValue{
+		Type:  uint64(tv.Type),
+		Uint:  omitEmpty(tv.Uint),
+		Bytes: sliceOrNil([]byte(tv.Bytes)),
```

Comment:
A global or local key name is allowed to be "" (empty).
Does this do the right thing? I think it might convert to `nil`.

Reply:
hmm, I think you are pointing at wrong place? state key is converted at line 410?

Reply:
Yes, totally right. But I suppose it's the same issue. A key's value can be set to "". `sliceOrNil` gives a `nil` when len(s) == 0.

So if someone sets a state value to empty string, we get a nil.

Reply:
I guess that's ok? We'll elide the `Bytes` value, but since we said it was of type byte, the debugger can deduce it's the empty string?

Reply:
yes, I was thinking that, the default omitted value get deduced to empty string, and I don't think we allow nil key in states, so this should not be an issue for debugger side.